### PR TITLE
firststrike tests

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg [hunk 1]
```diff
@@ -0,0 +1,101 @@
+#textdomain wesnoth-test
+
+#####
+# API(s) being tested: [firststrike]
+##
+# Actions:
+# Give bob a firststrike ability
```
Comment (ProditorMagnus) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:7]: (for multiple tests) Why it does 2 [test_do_attack_by_id] between same units. Looks like the only thing second tests is [modify_unit] affecting all weapons.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+            [/filter]
+            [effect]
+                apply_to = new_ability
+                [abilities]
+                    {TEST_ABILITY_NO_VALUE firststrike () SELF=yes}
+                [/abilities]
+            [/effect]
+        [/modify_unit]
+
+        [modify_unit]
```
Comment (ProditorMagnus) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:30]: candidate for macro with [filter][/filter]

Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:30]: What do you mean?

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:7]: They're using weapon 0 and then weapon 1.

Comment (ProditorMagnus) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:7]: But both of them have first strike.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg (hunk 2 above)
Comment (ProditorMagnus) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:30]: Looked like code used for multiple tests.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:7]: The filter gives `id = bob`.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/firststrike/firststrike_ability.cfg:30]: It's only for a few tests though, specific to firststrike. Or at least that was my thinking.

**Pull request merged**